### PR TITLE
check-sex query changes including documentation.

### PR DESCRIPTION
**Pull request by arostamianfar**
@deflaux 

Updated check-sex query to use the same format as homozygous-variants.sql. Also, added documentation explaining the difference with PLINK results.

### CODE REVIEW

File: R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      contain merged info about all samples for each SNP. Thus, using
+      separate VCF files (each containing info for a specific individual)
+      was not feasible.
+
+  * [chrX-all-samples-merged.fam](chrX-all-samples-merged.fam):
+    * This file contains pedigree information for all 17 individuals.
+
+#### Outputs:
+  * [sex-check-with-fam](sex-check-with-fam.sexcheck):
+    * PLINK output when using actual pedgiree info.
```
Comment (pgrosu) [inline R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md]: s/pedgiree/pedigree

Comment (arostamianfar) [inline R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md]: Thanks for catching this! Fixed.

Comment (pgrosu) [inline R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md]: Sure thing - thanks :)

File: R/PlatinumGenomes-QC/sql/check-sex.sql [hunk 2]
```diff
@@ -1,35 +1,51 @@
...
-  HAVING
-    # Skip 1/2 genotypes _and non-SNP variants
-    num_alts = 1
-    AND reference_bases IN ('A','C','G','T')
-    AND alternate_bases IN ('A','C','G','T')
-  )
-GROUP BY
-  call.call_set_name
+    SUM(first_allele = second_allele) AS O_HOM,
+    SUM(1.0 - (2.0 * freq * (1.0 - freq) * (called_allele_count / (called_allele_count - 1.0)))) AS E_HOM,
```
Comment (deflaux) [inline R/PlatinumGenomes-QC/sql/check-sex.sql]: Is this query exactly the same as https://github.com/googlegenomics/codelabs/blob/master/R/PlatinumGenomes-QC/sql/homozygous-variants.sql except for the WHERE clause?

File: R/PlatinumGenomes-QC/comparison/QC-Comparison.Rmd [hunk 3]
```diff
@@ -40,6 +40,37 @@ replacements <- list("_GENOME_CALL_TABLE_"="genomics-public-data:platinum_genome
...
+
+The slight difference in the F values is due to differences in allele frequency calculations, which results in more SNPs being ignored in PLINK. For instance, ~85,000 of ~345,000 SNPs are excluded in PLINK but only ~41,000 are excluded in the above query (due to having frequency of 1.0 or 0.0). As a concrete example, the SNP in position 2704300 has 15 individuals with 0/0 allele and 2 individuals with 0/1 allele. The minor allele frequency is therefore 2/34 = 0.06, whereas the PLINK allele frequency is 1.0 (actual ratio in PLINK's calculations is 24/24), which results in exclusion of this SNP from the PLINK analysis.
+
+PLINK also considers pedigree info in its analysis. The above results do not use pedigree info, but those results are also provided in [sex-check-with-fam](./sexCheck/plinkCombined/sex-check-with-fam.sexcheck).
+
+Lastly, the 'expanded table' used in the query only contains non-variant segments for SNPs (i.e. indels matching the reference are not included). Thus, indels were also excluded from the above analysis as they would result in inaccurate statistics.
```
Comment (deflaux) [inline R/PlatinumGenomes-QC/comparison/QC-Comparison.Rmd]: similar comment about updated language

File: R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+#### Inputs:
+  * [chrX-all-samples-merged.bim](chrX-all-samples-merged.bim) and [chrX-all-samples-merged.bed](chrX-all-samples-merged.bed):
+    * These files were transformed from a VCF file that was created using
+      the query included below. The primary reason for using a customized input
+      file was to ensure PLINK used the same input as BigQuery. For instance,
+      the 'expanded table' used in the query only contains non-variant segments
+      for SNPs (i.e. indels matching the reference are not included). Thus,
+      indels needed to be excluded in the provided inputs.
+      Also, to obtain accurate results in PLINK, the input files must
```
Comment (deflaux) [inline R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md]: We're using updated language now `expanded table` -> `multi-sample variants table` per http://biorxiv.org/content/early/2015/12/24/035295

Also, the multi-sample variants table does not include any non-variant segments.  What it does include are reference calls for a SNP for any samples with a non-variant segment that overlapped that SNP.

Suggested edit:
*For instance, the multi-sample variants table used in the query only contains reference-matches for SNPs (i.e. indels matching the reference are not included).*

Comment (pgrosu) [inline R/PlatinumGenomes-QC/comparison/sexCheck/plinkCombined/README.md]: Congratulations on the publication!

**Comment by deflaux** [comment]
@arostamianfar nice work here!

I'm leaning towards keeping the existing sex-check query and adding this as a second one.  Let me know about the overlap with the other query and we'll decide how to proceed.

File: R/PlatinumGenomes-QC/sql/check-sex.sql (hunk 2 above)
Comment (pgrosu) [inline R/PlatinumGenomes-QC/sql/check-sex.sql]: Yup :)

```
$ diff homozygous-variants.sql check-sex.sql
1c1,2
< # Compute the expected and observed homozygosity rate for each individual.
---
> # Compute homozygosity (inbreeding) rate on chromosome X for each individual,
> # to help determine the correctness of the sex phenotype value.
30,32c31,35
<     # Optionally add a clause here to limit the query to a particular
<     # region of the genome.
<     #_WHERE_
---
>     WHERE
>       (reference_name = 'chrX' OR reference_name = 'X')
>       # Omit pseudoautosomal regions.
>       AND start NOT BETWEEN 59999 AND 2699519
>       AND start NOT BETWEEN 154931042 AND 155260559
$
```

**Comment by arostamianfar** [comment]
@deflaux  Thanks for the review! I updated the explanations to mention the new table name.

About the query: yes, as Paul also mentioned, the modified sex-check query is exactly the same as homozygous-variants.sql except for the where clause. The reason I decided to take this approach was to be consistent with the logic in PLINK and also to show how the same query can be re-used for two different purposes (maximizing code reuse principle :) ). However, please let me know how you like to proceed with this query.

**Comment by deflaux** [comment]
Thank you @arostamianfar !!

**Pull request merged**